### PR TITLE
[Merged by Bors] - feat: port Topology.Bornology.Basic

### PR DESCRIPTION
**Pull request by j-loreaux**
porting notes:

1. There were two places where Lean couldn't figure out the type class (`Bornology`) arguments to `isBounded_def` when used within a rewrite or simp lemma, and I had to provide these explicitly. I have worked around it for now (by supplying those arguments), but Lean 3 did not have this problem and we should investigate the issue.
2. Because Lean 4 does not accept the `[]` syntax in structure fields to make arguments explicit, it was necessary to rename the structure fields and then provide new `Bornology.cobounded` and `Bornology.le_cofinite` declarations with the type argument explicit. See the discussion here: https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/Topology.2EBornology.2EBasic

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by jcommelin** [comment]
@j-loreaux Could you please open a separate PR for each file. That way it is a lot easier to review the PR against the baseline output of mathport.
I think you can just run `start_port.sh` for the 3 new PRs, and copy the output from this PR to those new PRs. Then they will all have 1 extra commit, besides the commits generated by `start_port.sh`. And reviewers can than easily review that single commit.

**Comment by j-loreaux** [comment]
Sure, sorry. I thought we were still on the "if you can fit more files into a PR in a natural way then do it" strategy.

**Comment by j-loreaux** [comment]
I rebased onto the latest master and force pushed to make the history clean. This is the diff you will want to see: https://github.com/leanprover-community/mathlib4/compare/dafdb9f98b896172b330a86afeb57465430ad302..24fa4adeeef18b91c671e18a3d45d0372a048539

File: Mathlib/Topology/Bornology/Basic.lean [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+The specification of a bornology in terms of the cobounded filter is equivalent to the standard
+one (e.g., see [Bourbaki, *Topological Vector Spaces*][bourbaki1987], **covering bornology**, now
+often called simply **bornology**) in terms of bounded sets (see `Bornology.ofBounded`,
+`IsBounded.union`, `IsBounded.subset`), except that we do not allow the empty bornology (that is,
+we require that *some* set must be bounded; equivalently, `∅` is bounded). In the literature the
+cobounded filter is generally referred to as the *filter at infinity*.
+
+## Main definitions
+
+- `Bornology α`: a class consisting of `cobounded : filter α` and a proof that this filter
```
Comment (jcommelin) [inline Mathlib/Topology/Bornology/Basic.lean:31]:
```suggestion
- `Bornology α`: a class consisting of `cobounded : Filter α` and a proof that this filter
```
Note that you can use `scripts/fix-comments.py File.Name.lean` to fix aligned names in comments.

**Review by jcommelin — COMMENTED** [review]
Thanks!

bors d+

File: Mathlib/Topology/Bornology/Basic.lean (hunk 1 above)
Comment (j-loreaux) [reply Mathlib/Topology/Bornology/Basic.lean:31]: oh cool! I didn't know about that.

**Pull request closed without merging**